DragaliaAPI/Services/UpdateDataService.cs

```diff
@@ -29,9 +29,36 @@ public UpdateDataList GetUpdateDataList(string deviceAccountId)
             )
             .Select(x => x.Entity);
 
+        Task<long> t = apiContext.PlayerWallet
+            .Where(
+                x =>
+                    x.DeviceAccountId == deviceAccountId
+                    && x.CurrencyType == Shared.Definitions.Enums.CurrencyTypes.FreeDiamantium
+            )
+            .Select(x => x.Quantity)
+            .FirstOrDefaultAsync();
+        t.Wait();
+
+        Task<long> t2 = apiContext.PlayerWallet
+            .Where(
+                x =>
+                    x.DeviceAccountId == deviceAccountId
+                    && x.CurrencyType == Shared.Definitions.Enums.CurrencyTypes.PaidDiamantium
+            )
+            .Select(x => x.Quantity)
+            .FirstOrDefaultAsync();
+        t2.Wait();
+
+        DiamondData diaData = new DiamondData()
+        {
+            free_diamond = (int)t.Result,
+            paid_diamond = (int)t2.Result
+        };
+
```

Comment:
The UpdateDataService should not be performing queries and waiting for them. The point is to query the change tracker, not the actual database itself. This implementation will return diamond_data on every single update_data_list